contracts/PartyToken.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+// ============ External Imports: Inherited Contracts ============
+import {ERC20VotesComp} from "@openzeppelin/contracts/token/ERC20/extensions/ERC20VotesComp.sol";
+import {ERC20Permit} from "@openzeppelin/contracts/token/ERC20/extensions/draft-ERC20Permit.sol";
+import {ERC20} from "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {Pausable} from "@openzeppelin/contracts/security/Pausable.sol";
+
+/*
+Party Token
+by Anna Carroll
+*/
+contract PartyToken is ERC20VotesComp, Pausable {
+    // ============ Immutables ============
+
+    address public immutable partyDAOMultisig;
+    address public immutable deprecationContract;
+
+    // ======== Constructor =========
+
+    constructor(address _partyDAOMultisig, address _deprecationContract) ERC20("PartyDAO", "PARTY") ERC20Permit("PartyDAO") {
+        // set partyDAO multisig & deprecation contract addresses
+        partyDAOMultisig = _partyDAOMultisig;
+        deprecationContract = _deprecationContract;
+        // mint 10M totalSupply to partyDAO multisig
+        _mint(_partyDAOMultisig, 10_000_000 * (10 ** 18));
+        // pause transfers; unpaused when lockup ends
+        _pause();
+    }
+
+    // ======== External Functions =========
+
+    function endLockup() external {
+        require(msg.sender == partyDAOMultisig, "only partyDAO");
+        _unpause();
+    }
+
+    // ======== Public Functions =========
+
+    /**
+     * @dev See {IERC20-transfer}.
+     * Requirements:
+     * - `recipient` cannot be the zero address.
+     * - the caller must have a balance of at least `amount`.
+     * - the contract cannot be paused
+     */
+    function transfer(address recipient, uint256 amount) public override whenNotPaused returns (bool) {
+        _transfer(_msgSender(), recipient, amount);
+        return true;
+    }
+
+    /**
+     * @dev See {IERC20-transferFrom}.
+     * Emits an {Approval} event indicating the updated allowance. This is not
+     * required by the EIP. See the note at the beginning of {ERC20}.
+     * Requirements:
+     * - `sender` and `recipient` cannot be the zero address.
+     * - `sender` must have a balance of at least `amount`.
+     * - the caller must have allowance for ``sender``'s tokens of at least
+     * `amount`.
+     * - the contract cannot be paused
+     */
+    function transferFrom(
+        address sender,
+        address recipient,
+        uint256 amount
+    ) public override whenNotPaused returns (bool) {
+        _transfer(sender, recipient, amount);
+
+        uint256 currentAllowance = allowance(sender, _msgSender());
+        require(currentAllowance >= amount, "ERC20: transfer amount exceeds allowance");
+        unchecked {
```

Comment:
honestly it actually seems wrong to me, for two reasons: 

first, 
> The setting only affects the statements that are syntactically inside the block. Functions called from within an unchecked block do not inherit the property.

it seems like the call to the `_approve` function doesn't inherit the property. 

second, there is no arithmetic being performed in the `_approve` function anyway, only an assignment 🤔

Reply:
ah, chatting to a friend, it seems that they did this so that they could make the explicit check themselves with a custom error message, then disable doing the same check again on the arithmetic to save gas.